resources_portal/models/material_request.py

```diff
@@ -0,0 +1,38 @@
+from django.db import models
+
+from .material import Material
+from .user import User
+
+
+class MaterialRequest(models.Model):
+    class Meta:
+        db_table = "material_requests"
+        get_latest_by = "created_at"
+
+    objects = models.Manager()
+
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    material = models.ForeignKey(
+        Material, blank=False, null=False, on_delete=models.CASCADE, related_name="requests"
+    )
+
+    requester = models.ForeignKey(
+        User, blank=False, null=False, on_delete=models.CASCADE, related_name="material_requests"
+    )
+
+    assigned_to = models.ForeignKey(
+        User, blank=False, null=False, on_delete=models.CASCADE, related_name="assigned_to_set"
+    )
+
+    mta = models.CharField(max_length=255, blank=True, null=True)
+    irb = models.CharField(max_length=255, blank=True, null=True)
```

Comment:
I think both of these fields are supposed to be PDFs.

Reply:
Are we uploading them to the DB with a [FileField](https://docs.djangoproject.com/en/3.0/ref/models/fields/#filefield)? I thought we would upload them to S3 and store the urls here.

Reply:
Hmm, I like that idea more! Good thought! Can you add a comment or maybe just change the field names to `mta_s3_url` and `irb_s3_url`?

Reply:
In that case, do we need to have `s3_bucket` and `s3_key` for each one of these fields? These are the arguments required by [S3.upload_file](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.upload_file).

Reply:
Hmm, the alternative would be to store the HTTPS address and then just create a wrapper function for `S3.upload_file` that parses out the bucket and key from the URL. Then we only need one field.

I'm fine with doing it either way, but I think it might make it easier to use our API if just stores/returns a URL.

Reply:
I like that, let's do it for all files. I'm also going to use a `TextField` for all URLs.

Reply:
Is there a reason not to use a URL field?

Reply:
URLfields are Charfields under the hood, so it needs to have some kind of max length. I think all it gets us is validation by a URLValidator, which I don't really think we need since we'll be creating and setting these S3 urls. I I think I'd rather not deal with an issue in the future where someone is trying to upload a really long filename and we're hitting the maxlength.

Reply:
Is there any security risk with allowing arbitrary filenames and then storing those? Would it be better to assign some sort of uuid or similar to uploaded files?

Reply:
Mmm, I think it depends on how we do things: https://owasp.org/www-community/vulnerabilities/Unrestricted_File_Upload

As far as I can tell, if we have the frontend actually just upload the file directly to S3 then we should be fine. I don't really care whether we let the user control the name or not, but I think it should be contextual and not a UUID. For example if we wanted to we could have templates like `material_request_{ID}_requester_signed_MTA.pdf` and `material_request_{ID}_executed_MTA.pdf`.